### PR TITLE
Implements log rotation in the Thermos runner.

### PR DESCRIPTION
**Pull request by GeorgeSirois**
@steveniemitz @chuegle @dmontauk @ThanosBaskous 

We'll need to discuss how we want to get this merged in, but it's a complete implementation in the current state.

### CODE REVIEW

**Comment by steveniemitz** [comment]
(thumbsup)

File: src/main/python/apache/thermos/core/process.py [hunk 1]
```diff
@@ -80,6 +82,8 @@ def __init__(self, name, cmdline, sequence, pathspec, sandbox_dir, user=None, pl
       optional:
         user        = the user to run as (if unspecified, will default to current user.)
                       if specified to a user that is not the current user, you must have root access
+        log_maxbytes = The maximum size of the stdout/stderr logs.
```
Comment (chuegle) [inline src/main/python/apache/thermos/core/process.py]: nit: I think it's preferred to align the = in doc.

File: src/main/python/apache/thermos/core/process.py [hunk 2]
```diff
@@ -348,19 +349,31 @@ def execute(self):
...
-                                   env=env)
+    subprocess_args = {
+      'args': ["/bin/bash", "-c", self.cmdline()],
+      'close_fds': self.FD_CLOEXEC,
+      'cwd': sandbox,
+      'env': env,
+      'pathspec': self._pathspec
+    }
+
+    if self._log_maxbackups > 0 and self._log_maxbytes > 0:
```
Comment (chuegle) [inline src/main/python/apache/thermos/core/process.py]: Should self._log_maxbackups default to 1 if self._log_maxbytes > 0 and self._log_maxbackups == 0?

File: src/main/python/apache/thermos/core/process.py (hunk 1 above)
Comment (GeorgeSirois) [inline src/main/python/apache/thermos/core/process.py]: Fixed this + other instances.

File: src/main/python/apache/thermos/core/process.py (hunk 2 above)
Comment (GeorgeSirois) [inline src/main/python/apache/thermos/core/process.py]: Yeah, that's probably a sane default.

**Comment by ThanosBaskous** [comment]
lgtm

**Pull request closed without merging**